worker/dockerworker/dockerworker.go

```diff
@@ -93,23 +94,35 @@ func (d *dockerworker) StartWorker(state *run.State) (protocol.Transport, error)
 		return nil, fmt.Errorf("Error writing worker config to %s: %v", d.wicfg.ConfigPath, err)
 	}
 
+	transp := protocol.NewStdioTransport()
+
 	// the --host taskcluster-worker-runner instructs docker-worker to merge
 	// config from $DOCKER_WORKER_CONFIG.
 	mainJs := fmt.Sprintf("%s/src/bin/worker.js", d.wicfg.Path)
 	cmd := exec.Command("node", mainJs, "--host", "taskcluster-worker-runner", "production")
 	cmd.Env = os.Environ()
 	cmd.Env = append(cmd.Env, "DOCKER_WORKER_CONFIG="+d.wicfg.ConfigPath)
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
+	cmd.Stdout = transp
 	cmd.Stderr = os.Stderr
 	d.cmd = cmd
 
+	// Unfortunately, cmd.Wait does not handle the case where cmd.Stdin is a writer that remains
+	// open when the process exits.  Instead, we set up our own copy loop, which will exit when
+	// the worker exits
+	pipe, err := cmd.StdinPipe()
+	if err != nil {
+		return nil, err
+	}
+	go func() {
+		_, _ = io.Copy(pipe, transp)
```

Comment:
I saw that this was originally `go io.Copy()` - is that a lint fail? Should we log the error here?

Reply:
Maybe.. I think we'll always get an error when `cmd.Wait` closes the pipe, and in any other case I'm not sure what we can do.  I'll see if I can filter out that case and at least log the others.

Reply:
Oh, interesting, that copy loop never exits, I suppose because it's reading from `transp` which never closes.  Concurrent programming is hard!

Reply:
I looked into fixing this never-ending copy.  The fix would be to close the `transp.Out` channel.  However, that leaves a bunch of race conditions, where some other part of tc-worker-runner has a timer running and tries to send a message.  Putting the message into `transp.Out` will then panic.  One fix to *that* is to add a `closed` property to the transport, but then to avoid data races that needs its own `sync.Mutex`.  Then we have two mutexes in the struct (and thus potential for deadlock), plus we have to figure out what to do with messages that are sent to a closed transport.

So I think we'll just live with the never-ending copy.  I added some logging to the `io.Copy` on error.